src/sql/workbench/contrib/welcome/gettingStarted/browser/initialSetupWizard.ts

```diff
@@ -0,0 +1,484 @@
+			this.hide();
+			return;
+		}
+		let next: number = i + 1;
+		const h1: HTMLElement = popups[next].querySelector('.ads-initial-setup-wizard-popup-content-container h1');
+		popups[i].classList.add('ads-initial-setup-wizard-hide');
+		popups[i].classList.remove('ads-initial-setup-wizard-show');
+		popups[i].classList.remove('ads-initial-setup-animate');
+		popups[next].classList.add('ads-initial-setup-wizard-show');
+		popups[next].classList.remove('ads-initial-setup-wizard-hide');
+		h1.focus();
+	}
+
+	private wizardPrevEvent(popups: NodeListOf<Element>, popupsLength: number, i: number): void {
+		let last: number = i - 1;
+		const h1: HTMLElement = popups[last].querySelector('.ads-initial-setup-wizard-popup-content-container h1');
+		popups[i].classList.add('ads-initial-setup-wizard-hide');
+		popups[i].classList.remove('ads-initial-setup-wizard-show');
+		popups[last].classList.add('ads-initial-setup-wizard-show');
+		popups[last].classList.remove('ads-initial-setup-wizard-hide');
+		h1.focus();
+	}
+
+
+	private buildInteractions(): void {
+		const popups = document.querySelectorAll('.ads-initial-setup-wizard-popup') as NodeListOf<Element>;
+		const menuBarItems = document.querySelectorAll('.menubar-menu-button') as NodeListOf<HTMLElement>;
+		const context: this = this;
+		menuBarItems.forEach((elm) => {
+			elm.style.pointerEvents = 'none';
+		});
+		popups.forEach(function (elm, i) {
+			const nextBtn = elm.querySelector('.ads-initial-setup-wizard-next-btn-container .monaco-button') as HTMLElement;
+			const previousBtn = elm.querySelector('.ads-initial-setup-wizard-previous-btn-container .monaco-button') as HTMLElement;
+			const popupsLength: number = popups.length;
+
+			dom.addStandardDisposableListener(nextBtn, 'keydown', event => {
+				if (event.equals(KeyCode.Enter)) {
+					context.wizardNextEvent(popups, popupsLength, i);
+				}
+				if (event.equals(KeyCode.Tab)) {
+					event.preventDefault();
+				}
+			});
+
+			dom.addStandardDisposableListener(nextBtn, 'click', event => {
+				context.wizardNextEvent(popups, popupsLength, i);
+			});
+
+			if (previousBtn) {
+
+				dom.addStandardDisposableListener(previousBtn, 'keydown', event => {
+					if (event.equals(KeyCode.Enter)) {
+						context.wizardPrevEvent(popups, popupsLength, i);
+					}
+				});
+
+				dom.addStandardDisposableListener(previousBtn, 'click', event => {
+					context.wizardPrevEvent(popups, popupsLength, i);
+				});
+			}
+		});
+		this.show();
+	}
+
+	public show(): void {
+		if (this._overlay.style.display !== 'block') {
+
+			const firstSetupElement = document.querySelector('.ads-initial-setup-wizard-element') as HTMLDivElement;
+			const setupWizard = document.querySelector('.ads-initial-setup-wizard') as HTMLElement;
+
+			setTimeout(() => {
+				setupWizard.classList.add('open');
+				firstSetupElement.classList.add('ads-initial-setup-animate');
+				firstSetupElement.classList.add('ads-initial-setup-wizard-show');
+			}, 1000);
```

Comment:
why is this delay needed and how was the duration determined?

Reply:
@kburtram This delay is for visual effect only, this was timing was reviewed by myself and UX. Without the timing the setup wizard will be launched the moment the application launches. UX wanted users to know that this is an overlay on top of the welcome page, so there is a slight delay before the overlay/setup wizard animation.